### ISSUE TITLE
Modal no longer respects enableAriaHiddenSiblings causing a11y issues

### ISSUE THREAD

**Issue by stefaniehansen** [microsoft/fluentui#21091]
### Environment Information

- **Package version(s)**:  @fluentui/react: 8.48.0

### Please provide a reproduction of the bug in a codepen:
https://codepen.io/stefuhnee/pen/MWEmzGR - check the main div with the id of "content".

#### Actual behavior:
enableAriaHiddenSiblings doesn't do anything for the Modal component. It looks like #17925 / [this commit](https://github.com/microsoft/fluentui/commit/83c7ebbe3aae5cb052e4f1405cbc8d68dc79aa8f#diff-3bfe049417a04943844a9c74bb54bc8caabf3d2acadd8e20a1747a2063696040) changed the behavior of Modal by affecting Popup. It also affects all other components that use Popup.

#### Expected behavior:
enableAriaHiddenSiblings should be respected for Modal and added for other components that use Popup internally. The default behavior should not have changed without an option to opt out or perhaps a new major version if Fluent uses Semver.

### Priorities and help requested:
Are you willing to submit a PR to fix? Yes, if I have time.

Requested priority: High - this is causing an accessibility bug for us that will affect our MAS rating.  Our Announced component is not rendered inside of the Modal, and we do not want to move it inside the Modal in case the Modal unmounts before it can announce a status change. Expected statuses are no longer announced. We forced an older version of Fluent inside OWA for our integration, but maintaining duplicate dependencies is not a good solution.

Products/sites affected: OneNote Feed integration in OneNote clients and OWA. The Notes Folder in OWA.

**Comment by ecraig12345** [comment]
Sorry about the breaking change! For future reference, for scorecard-impacting bugs it's better to file an issue via our [internal process](https://aka.ms/fluentui-a11y) to ensure it's addressed within your SLA, and so it can be directly linked to your bug.

Based on the prop comment and git history, it looks like [`enableAriaHiddenSiblings`](https://github.com/microsoft/fluentui/search?q=enableAriaHiddenSiblings) was added in version 7 and intended to become the default behavior in version 8, but we missed that while working on the version 8 release. Since it's definitely a breaking change and missed the release, we'd typically leave the prop and behavior as-is for the duration of version 8 unless there was an extremely compelling need to change it.
https://github.com/microsoft/fluentui/blob/2fa6e29384ee32c3d5bb0f84ea05c2ad648c37f5/packages/react/src/components/FocusTrapZone/FocusTrapZone.types.ts#L90-L95

The thing that's a bit odd about #17925 (and follow-up #19628) is that it adds an *additional* mechanism in Popup for hiding siblings instead of using `enableAriaHiddenSiblings` which is ultimately handled in FocusTrapZone. @TristanWatanabe might be able to provide more background about why it was implemented that way, or maybe it was an oversight.

One possible solution is like you implemented in #21110, adding `enableAriaHiddenSiblings` in more places and moving the new sibling-hiding mechanism in Popup under that prop. However, that doesn't ultimately solve the problem. I'm guessing in your case, it's still desirable for everything *except* the Announced to be hidden. And after hiding siblings becomes the default, there needs to be a way to opt out of this behavior for specific individual components.

A better solution might be to

**Comment by ecraig12345** [comment]
Somehow submitted the previous comment too soon, but now it's updated with the rest of what I meant to write. Also to set expectations, any fix here would have significant enough API and behavior impact that it should be discussed with a couple team members first, and that may not happen until the new year due to people being out for the holidays (I'm not sure who's around today or next week).

**Comment by stefaniehansen** [comment]
That makes sense to me - thanks for taking a look and thinking through the problem. Your proposed approach makes sense to me, although I'll leave it to the folks more familiar with working on Fluent code to discuss :) It would be ideal if we could take a look at this in January, if possible, to avoid OWA having to pull in two different versions of Fluent. I'll leave my PR hanging around for now, although I agree that there are better approaches.

**Comment by stefaniehansen** [comment]
@ecraig12345, should I create a bug via the internal process that you linked or is this sufficient for this issue? Thanks for letting me know about that process - I'll certainly use it in the future.

**Comment by stefaniehansen** [comment]
@ecraig12345 - a friendly ping on this after the holidays!

**Comment by smhigley** [comment]
This is an interesting issue, because I'm Fluent providing any sort of opt-out for `aria-hidden` on elements outside an open modal. The reason comes from the way dialogs and modals are designed to function at a language and browser level, even if support isn't currently there (see: https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog and the 4th paragraph here: https://w3c.github.io/aria/#aria-modal).

If we explicitly enable opting out of hiding elements outside an open modal, we're going against how that pattern is spec'd to behave, and limiting our ability to switch to platform features as they become available. We could still do it, but it would be a lever for an antipattern, and I don't think we should carry it over to vNext.

That said, I do see the use case for wanting a stable non-modal DOM subtree if you're expecting to fire live regions right when the modal opens or closes. One solution would be to use lifecycle hooks (and maybe a timeout) to update the content of a live region when the modal opens or closes, but I definitely see how that would be messy.

I think maybe the best solution is to go ahead and provide an opt-out in v8, with the knowledge that we won't carry it over to vNext (I'd like to be able to use the dialog element there, so we can eventually take out all custom modal polyfills once browser support is there).

**Comment by stefaniehansen** [comment]
@smhigley - Thanks for the thoughtful response. That makes sense to me, and hopefully we can figure out a way a reasonable way to fire a live region when a modal opens or closes without hacky workarounds for vNext. 

If you have time, I'd love to get your feedback on my pull request linked to this issue. I'll be addressing Elizabeth's comments and pushing a new iteration soon.

**Comment by ecraig12345** [comment]
Talked to Sarah again and we're basically going with what she proposed in her most recent comment: we'll use @stefaniehansen's approach in #21110 for v8, but due to upcoming platform features there will need to be a different solution for v9.
